src/main/java/ca/gc/aafc/collection/api/validation/GeoreferenceAssertionValidator.java

```diff
@@ -7,31 +7,37 @@
 import org.springframework.validation.Validator;
 
 import ca.gc.aafc.collection.api.entities.GeoreferenceAssertion;
+import ca.gc.aafc.collection.api.entities.GeoreferenceAssertion.GeoreferenceVerificationStatus;
 
 @Component
 public class GeoreferenceAssertionValidator implements Validator {
 
-    private MessageSource messageSource;
+  private MessageSource messageSource;
 
-    public GeoreferenceAssertionValidator(MessageSource messageSource) {
-        this.messageSource = messageSource;
-    }
-
-    @Override
-    public boolean supports(Class<?> clazz) {
-        return GeoreferenceAssertion.class.isAssignableFrom(clazz);
-    }
+  public GeoreferenceAssertionValidator(MessageSource messageSource) {
+    this.messageSource = messageSource;
+  }
 
-    @Override
-    public void validate(Object target, Errors errors) {
-        GeoreferenceAssertion georeferenceAssertion = (GeoreferenceAssertion) target;
+  @Override
+  public boolean supports(Class<?> clazz) {
+    return GeoreferenceAssertion.class.isAssignableFrom(clazz);
+  }
 
-        if (georeferenceAssertion.getCollectingEvent() != null && georeferenceAssertion.getCollectingEvent().getPrimaryGeoreferenceAssertion() == null) {
-            String errorMessage = messageSource.getMessage("georeferenceAssertion.collectingEvent.primaryGeoreferenceAssertion.null",
-                null, LocaleContextHolder.getLocale());
-            errors.rejectValue("collectingEvent", "georeferenceAssertion.collectingEvent.primaryGeoreferenceAssertion.null", errorMessage);
+  @Override
+  public void validate(Object target, Errors errors) {
+    GeoreferenceAssertion georeferenceAssertion = (GeoreferenceAssertion) target;
```

Comment:
We should probably check that the target is class is supported before casting. that support method seems useful in this case.

Reply:
that is what the isASsignableFrom in the supports method currently does 
https://docs.oracle.com/javase/7/docs/api/java/lang/Class.html#isAssignableFrom%28java.lang.Class%29

Reply:
the supports method is run before the validate method

Reply:
This appears to be called directly from the CollectingEventService on line 62, additionaly I would be a bit worried as anybody can inject this into a class and call it directly.

Probably safe to just check before we cast.